CMakeLists.txt

```diff
@@ -599,6 +580,9 @@ if(DEFINED ENV{LDFLAGS})
   deduplicate_flags(CMAKE_EXE_LINKER_FLAGS)
 endif()
 
+if(BUILD_TESTS)
```

Comment:
Why is this needed for secp256k1 specifically?

Reply:
It is not secp256k1-specific change. As a part of the "[FIXUP] cmake: Enable tests before adding subdirectories" commit this change adjusts the moment when `enable_testing()` being called. Previously, it was done implicitly via `include(CTest)` command _after_ processing the `src` subdirectory`. Now it done explicitly and _before_ processing the `src` subdirectory. Which activates any `add_test` commands within the `src` subdirectory including the `src/secp256k1` subdirectory as well.